api/v1alpha1/obtenant_types.go

```diff
@@ -17,29 +17,114 @@ limitations under the License.
 package v1alpha1
 
 import (
+	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized
 
 // OBTenantSpec defines the desired state of OBTenant
 type OBTenantSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// Foo is an example field of OBTenant. Edit obtenant_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	ClusterName string `json:"clusterName"`
+	TenantName  string `json:"tenantName"`
+	UnitNumber  int    `json:"unitNum"`
+
```

Comment:
suggest to add a config for root password

Reply:
If you want to add a finalizer to a task, either execute the Update method on the task thread, causing additional CAS Update failures to be introduced, or add the execution flow of the Update method on the tuning control thread.

So I didn't add the finalizer as a separate task. Maybe it can be optimized in future commits.